src/Instructions.cpp

```diff
@@ -2125,7 +2132,15 @@ void Instructions::_goto_w(){
 }
 
 void Instructions::_jsr_w(){
-    addToPC(1);
+    Frame f = frames.top();
+    u1 branchbyte1 = f.bytecode[f.PC+1];
+    u1 branchbyte2 = f.bytecode[f.PC+2];
+    u1 branchbyte3 = f.bytecode[f.PC+3];
+    u1 branchbyte4 = f.bytecode[f.PC+4];
```

Comment:
Aquela parada lá dos bytes perderem os bits quando for fazer os shifts, dá pra mudar os tipos pra u4.